src/PetClass.js

```diff
@@ -0,0 +1,81 @@
+const petConfig = {
+  AGE_INIT: 0,
+  AGE_MAX: 30,
+  AGE_INCREMENT: 1,
+  FITNESS_INIT: 10,
+  FITNESS_MIN: 0,
+  FITNESS_INCREMENT: 4,
+  FITNESS_DECREMENT: 3,
+  FITNESS_THRESHOLD: 3,
+  HUNGER_INIT: 0,
+  HUNGER_MAX: 10,
+  HUNGER_INCREMENT: 5,
+  HUNGER_DECREMENT: 3,
+  HUNGER_THRESHOLD: 5
+};
+
+class Pet {
+  constructor(name) {
+    if (!(typeof name === 'string' || name === undefined)) {
+      throw new TypeError('name must be a string or undefined');
+    }
+    this.name = name;
+    this.age = petConfig.AGE_INIT;
+    this.fitness = petConfig.FITNESS_INIT;
+    this.hunger = petConfig.HUNGER_INIT;
+  }
+
+  growUp() {
+    if (!this.isAlive) throw 'Your pet is no longer alive';
+    this.age += petConfig.AGE_INCREMENT;
+    this.hunger += petConfig.HUNGER_INCREMENT;
+    this.fitness -= petConfig.FITNESS_DECREMENT;
```

Comment:
Does there need to be any checks that any of these values wont go beyond their maximum values for he pet to stay alive?